crates/nostr-mls-memory-storage/src/groups.rs

```diff
@@ -0,0 +1,128 @@
+//! Memory-based storage implementation of the NostrMlsStorageProvider trait for Nostr MLS groups
+
+use std::sync::Arc;
+
+use nostr::PublicKey;
+use nostr_mls_storage::groups::error::GroupError;
+use nostr_mls_storage::groups::types::*;
+use nostr_mls_storage::groups::GroupStorage;
+use nostr_mls_storage::messages::types::Message;
+
+use crate::NostrMlsMemoryStorage;
+
+impl GroupStorage for NostrMlsMemoryStorage {
+    fn save_group(&self, group: Group) -> Result<Group, GroupError> {
```

Comment:
I missed this in the previous PR, I think the `save_*` methods shouldn't return the same value passed as argument. IMO here it should return a `Result<(), GroupError>`.

What do you think? If you agree, I can change this in another PR.